### ISSUE TITLE
xsd2json: Attribute appears twice in restrictions

### ISSUE THREAD

**Issue by vog** [UNC-Libraries/jquery.xmleditor#82]
When an attribute of a `complexType` is restricted further, it appears twice in the output JSON.

In the following example, the attribute `foo` is optional in `BaseType`, but restricted to be required in `FooType`:

```xml
<schema xmlns:foo="http://example.com/foo" xmlns="http://www.w3.org/2001/XMLSchema" targetNamespace="http://example.com/foo" elementFormDefault="qualified">
  <complexType name="BaseType">
    <simpleContent>
      <extension base="string">
        <attribute name="foo" type="string" use="optional"/>
      </extension>
    </simpleContent>
  </complexType>
  <complexType name="FooType">
    <simpleContent>
      <restriction base="foo:BaseType">
        <attribute name="foo" type="string" use="required"/>
      </restriction>
    </simpleContent>
  </complexType>
  <element name="Main">
    <complexType>
      <sequence>
        <element name="Foo" type="foo:FooType"/>
      </sequence>
    </complexType>
  </element>
</schema>
```

Current output (relevant part):

```json
        {
          "values": [],
          "type": "string",
          "ns": 3,
          "name": "Foo",
          "element": true,
          "attributes": [
            {
              "values": [],
              "type": "string",
              "ns": 3,
              "name": "foo",
              "attribute": true
            },
            {
              "values": [],
              "type": "string",
              "ns": 3,
              "name": "foo",
              "attribute": true
            }
          ],
          "elements": []
        }
```

Expected output (relevant part):

```json
        {
          "values": [],
          "type": "string",
          "ns": 3,
          "name": "Foo",
          "element": true,
          "attributes": [
            {
              "values": [],
              "type": "string",
              "ns": 3,
              "name": "foo",
              "attribute": true
            }
          ],
          "elements": []
        }
```

**Comment by vog** [comment]
Provided test case in #83.

**Comment by vog** [comment]
While working on this issue, I discovered (and fixed) #86.

**Comment by vog** [comment]
It seems that this is more a feature implementation than a bugfix.

Right now, xsd2json treats all `restriction`s the same way as `extension`s, without any special handling.

**Comment by vog** [comment]
I won't be able to provide a full implementation of XML Schema restrictions, but will implement enough to solve the issues at hand.

**Comment by vog** [comment]
Discovered and fixed another bug on the go: #88.
